### PR TITLE
formatting multi_head_attention

### LINKED ISSUE #9639: multi_head_attention

### PR DESCRIPTION
**Pull request by abbaahmad**
close #9639

### CODE REVIEW

File: ivy/functional/ivy/layers.py [hunk 1]
```diff
@@ -728,9 +728,11 @@ def multi_head_attention(
         k, v = ivy.split(kv, num_or_size_splits=2, axis=-1)
 
     # BS x H x Q x F,  BS x H x K x F,  BS x H x K x F
+    def func(t):
+        return ivy.einops_rearrange(t,"... n (h f) -> ... h n f", h=num_heads)
```
Comment (Daniel4078) [inline ivy/functional/ivy/layers.py:732]: ivy/functional/ivy/layers.py:732:38: E231 missing whitespace after ','

**Review by Daniel4078 — CHANGES_REQUESTED** [review]
Hello, found one small formatting error for you to fix.

**Comment by Daniel4078** [comment]
By the way, please write a comment "add_reformatting_checklist_layers" here, so that you will get a checklist to accurately track your progresses.

**Comment by abbaahmad** [comment]
add_reformatting_checklist_layers

**Comment by abbaahmad** [comment]
@Daniel4078 I've made the requested changes. The linter also complained about a line in `run_tests.py` for which I made a change.  I, however, am not sure how to engage with the checklist for this particular task. Any help will be appreciated.

**Comment by Daniel4078** [comment]
> @Daniel4078 I've made the requested changes. The linter also complained about a line in `run_tests.py` for which I made a change. I, however, am not sure how to engage with the checklist for this particular task. Any help will be appreciated.

You can have a look at other pull requests that deal with a similiar issue like this one and see what their checklist looks like. In short, edit the existing legends of the checks use the given list of legends before to indicate the current status of the tasks discribed in the checks. In this way, I know what you have done and still need to do quickly.

**Comment by abbaahmad** [comment]
> ## Reformatting Task Checklist
> #### IMPORTANT NOTICE 🚨:
> The [Ivy Docs](https://lets-unify.ai/ivy/) represent the ground truth for the task descriptions and this checklist should only be used as a supplementary item to aid with the review process.
> 
> #### LEGEND 🗺:
> * ❌ :  Check item is not completed.
> * ✅ :  Check item is ready for review.
> * 🆘 :  Stuck/Doubting implementation (PR author should add comments explaining why).
> * ⏩ :  Check is not applicable to function (skip).
> * 🆗 :  Check item is already implemented and does not require any edits.
> 
> #### CHECKS 📑:
> 1. * [ ]  ✅:  Remove all lambda and direct bindings for the backend functions in:
>      
>      * [ ]  ⏩: [ivy/functional/backends/jax/layers.py](https://github.com/unifyai/ivy/tree/master/ivy/functional/backends/jax/layers.py).
>      * [ ]  ⏩: [ivy/functional/backends/numpy/layers.py](https://github.com/unifyai/ivy/tree/master/ivy/functional/backends/numpy/layers.py).
>      * [ ]  ⏩: [ivy/functional/backends/tensorflow/layers.py](https://github.com/unifyai/ivy/tree/master/ivy/functional/backends/tensorflow/layers.py).
>      * [ ]  ⏩: [ivy/functional/backends/torch/layers.py](https://github.com/unifyai/ivy/tree/master/ivy/functional/backends/torch/layers.py).
> 2. * [ ]  ⏩: Implement the following if they don't exist:
>      
>      1. * [ ]   ⏩: The `ivy.Array` instance method in [ivy/array/layers.py](https://github.com/unifyai/ivy/blob/master/ivy/array/layers.py).
>      2. * [ ]   ⏩: The `ivy.Array` special method in [ivy/array/array.py](https://github.com/unifyai/ivy/blob/master/ivy/array/array.py).
>      3. * [ ]   ⏩: The `ivy.Array` reverse special method in [ivy/array/array.py](https://github.com/unifyai/ivy/blob/master/ivy/array/array.py).
>      4. * [ ]  ⏩: The `ivy.Container` static method in [ivy/container/layers.py](https://github.com/unifyai/ivy/blob/master/ivy/container/layers.py).
>      5. * [ ]  ⏩: The `ivy.Container` instance method in [ivy/container/layers.py](https://github.com/unifyai/ivy/blob/master/ivy/container/layers.py).
>      6. * [ ]   ⏩:  The `ivy.Container` special method in [ivy/container/container.py](https://github.com/unifyai/ivy/blob/master/ivy/container/container.py).
>      7. * [ ]   ⏩: The `ivy.Container` reverse special method in [ivy/container/container.py](https://github.com/unifyai/ivy/blob/master/ivy/container/container.py).
> 3. * [ ]  ⏩:  Make sure that the aforementioned methods are added into the correct category-specific parent class, such as  `ivy.ArrayWithElementwise`,  `ivy.ContainerWithManipulation`  etc.
> 4. * [ ]  ✅:  Correct all of the  [Function Arguments and the type hints](https://lets-unify.ai/ivy/deep_dive/10_function_arguments.html#function-arguments) for every function  **and**  its  _relevant methods_, including those you did not implement yourself.
> 5. * [ ]  ⏩: Add the correct  [Docstrings](https://lets-unify.ai/ivy/deep_dive/docstrings.html#docstrings)  to every function  **and**  its  _relevant methods_, including those you did not implement yourself. The following should be added:
>      
>      1. * [ ]  ⏩:    The function's [Array API standard](https://data-apis.org/array-api/latest/index.html) description in [ivy/functional/layers.py](https://github.com/unifyai/ivy/blob/master/ivy/functional/ivy/layers.py). If the function is not part of the Array API standard then a description of similar style should be added to the same file.
>           The following modifications should be made to the description:
>           
>           * [ ]  ⏩:  Remove type definitions in the `Parameters` and `Returns` sections.
>           * [ ]  ⏩:  Add `out` to the `Parameters` section if function accepts an `out` argument.
>           * [ ]  ⏩:  Replace `out` with `ret` in the `Returns` section.
>      2. * [ ]  ⏩:  Reference to docstring for ivy.function_name ([5.a](#ref1)) for the function description **and** modified `Parameters` and `Returns` sections as described in [the docs](https://lets-unify.ai/ivy/deep_dive/docstrings.html#docstrings) in:
>           
>           * [ ]  ⏩:  [ivy/array/layers.py](https://github.com/unifyai/ivy/blob/master/ivy/array/layers.py).
>           * [ ]  ⏩:  [ivy/container/layers.py](https://github.com/unifyai/ivy/blob/master/ivy/container/layers.py) (in the static and instance method versions).
>           * [ ]  ⏩:   [ivy/array/array.py](https://github.com/unifyai/ivy/blob/master/ivy/array/array.py) if the function has a special method  ( like `__function_name__` ).
>           * [ ]  ⏩:  [ivy/array/array.py](https://github.com/unifyai/ivy/blob/master/ivy/array/array.py) if the function has a reverse special method  ( like `__rfunction_name__` ).
>           * [ ]  ⏩: [ivy/container/container.py](https://github.com/unifyai/ivy/blob/master/ivy/container/container.py) if the function has a special method ( like `__function_name__` ).
>           * [ ]  ⏩:  [ivy/container/container.py](https://github.com/unifyai/ivy/blob/master/ivy/container/container.py) if the function has a reverse special method  ( like `__rfunction_name__` ).
> 6. * [ ]  🆗: Add thorough  [Docstring Examples](https://lets-unify.ai/ivy/deep_dive/docstring_examples.html#docstring-examples)  for every function  **and**  its  _relevant methods_  and ensure they pass the docstring tests.
>      **Functional Examples** in [ivy/functional/layers.py](https://github.com/unifyai/ivy/blob/master/ivy/functional/ivy/layers.py).
>      
>      1. * [ ]  🆗: Cover all possible variants for each of the arguments independently (not combinatorily).
>      2. * [ ]  🆗: Vary the values and input shapes considerably between examples.
>      3. * [ ]  🆗: Start out simple and get more complex with each example.
>      4. * [ ]  ⏩: Show an example with:
>           
>           * [ ]  ⏩: `out` unused.
>           * [ ]  ⏩: `out` used to update a new array y.
>           * [ ]  ⏩: `out` used to inplace update the input array x (if x has the same dtype and shape as the return).
>      5. * [ ]  ⏩: If broadcasting is relevant for the function, then show examples which highlight this.
>      
>      **Nestable Function Examples** in [ivy/functional/layers.py](https://github.com/unifyai/ivy/blob/master/ivy/functional/ivy/layers.py).
>      Only if the function supports nestable operations.
>      
>      6. * [ ]  ⏩:  Add an example that passes in an  `ivy.Container`  instance in place of one of the arguments.
>      7. * [ ]  ⏩:  Add an example passes in  `ivy.Container`  instances for multiple arguments.
>      
>      **Container Static Method Examples** in [ivy/container/layers.py](https://github.com/unifyai/ivy/blob/master/ivy/container/layers.py).
>      
>      8. * [ ]  ⏩: The example from point ([6.f](#ref2)) should be replicated, but added to the  `ivy.Container`  **static method** docstring in with  `ivy.<func_name>`  replaced with  `ivy.Container.static_<func_name>`  in the example.
>      9. * [ ]  ⏩: The example from point ([6.g](#ref3)) should be replicated, but added to the  `ivy.Container`  **static method** docstring, with  `ivy.<func_name>`  replaced with  `ivy.Container.static_<func_name>`  in the example.
>      
>      **Array Instance Method Example** in [ivy/array/layers.py](https://github.com/unifyai/ivy/blob/master/ivy/array/layers).
>      
>      10. * [ ]  🆗: Call this instance method of the  `ivy.Array`  class.
>      
>      **Container Instance Method Example** in [ivy/container/layers.py](https://github.com/unifyai/ivy/blob/master/ivy/container/layers.py).
>      
>      11. * [ ]  🆗: Call this instance method of the  `ivy.Container`  class.
>      
>      **Array Operator Examples** in [ivy/array/array.py](https://github.com/unifyai/ivy/blob/master/ivy/array/array.py).
>      
>      12. * [ ]  🆗: Call the operator on two  `ivy.Array`  instances.
>      13. * [ ]  🆗: Call the operator with an  `ivy.Array`  instance on the left and  `ivy.Container`  on the right.
>      
>      **Array Reverse Operator Example** in [ivy/array/array.py](https://github.com/unifyai/ivy/blob/master/ivy/array/array.py).
>      
>      14. * [ ]  ⏩: Call the operator with a  `Number`  on the left and an  `ivy.Array`  instance on the right.
>      
>      **Container Operator Examples** in [ivy/container/container.py](https://github.com/unifyai/ivy/blob/master/ivy/container/container.py).
>      
>      15. * [ ]  ⏩: Call the operator on two `ivy.Container` instances containing Number instances at the leaves.
>      16. * [ ]  ⏩: Call the operator on two `ivy.Container` instances containing `ivy.Array` instances at the leaves.
>      17. * [ ]  ⏩: Call the operator with an `ivy.Container` instance on the left and `ivy.Array` on the right.
>      
>      **Container Reverse Operator Example** in [ivy/container/container.py](https://github.com/unifyai/ivy/blob/master/ivy/container/container.py).
>      
>      18. * [ ]  ⏩: Following example in the [`ivy.Container.__radd__`](https://github.com/unifyai/ivy/blob/e28a3cfd8a4527066d0d92d48a9e849c9f367a39/ivy/container/container.py#L173) docstring, with the operator called with a `Number` on the left and an `ivy.Container` instance on the right.
>      
>      **Tests**
>      
>      19. * [ ]  🆗: Docstring examples tests passing.
>      20. * [ ]  ✅: Lint checks passing.

@Daniel4078 Should be done now

**Review by Daniel4078 — APPROVED** [review]
Well done.

**Pull request merged**